### DISCUSSION TITLE
issue: One input refreshes app on enter key, multiple inputs dont behave this way

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#6745] [Q&A] [answered]
### Version Number

7.12.1

### Codesandbox/Expo snack

https://codesandbox.io/s/nifty-babycat-f33nd?file=/src/App.js

### Steps to reproduce

1. Click into input
2. Hit Enter on keyboard
3. Watch app refresh

### Expected behaviour

e.preventDefault() should take effect regardless of number of inputs. Only one input behaves this way but if you add a second input it the Enter key does nothing.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fahadsohail482** [comment, marked as answer]
@chaserda I see you don't have any submit button in the form. 
A form element's default button is the first submit button in tree order whose form owner is that form element.
If the user agent supports letting the user submit a form implicitly (for example, on some platforms hitting the "enter" key while a text control is focused implicitly submits the form).
If the form has no submit (button/input), then the implicit submission mechanism must do nothing if the form has more than one field that blocks implicit submission.
If you want to change this default behaviour you have to use "handleSubmit" on onSubmit event of the form provided by react-hook-form. This will prevent the default behaviour in any case.
here is the implementation reference of "handleSubmit" function's where it is handled by react-hook-form. 
             
      const handleSubmit: UseFormHandleSubmit < TFieldValues > =
           (onValid, onInvalid) => async (e) => {
                   if (e) {
                     e.preventDefault && e.preventDefault();
                     e.persist && e.persist();
                   }

[Updated CodeSandBox](https://codesandbox.io/s/solitary-moon-c60bd?file=/src/App.js)
[Reference](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#implicit-submission)
I hope this answer helps you.

**Comment by bluebill1049** [comment]
thanks @fahadsohail482 awesome answer and correct one too. 👍

**Reply by fahadsohail482** [reply]
You're welcome @bluebill1049.
